src/main/scala/firrtl/passes/LowerTypes.scala

```diff
@@ -139,7 +139,7 @@ object LowerTypes extends Pass {
       def lowerTypesMemExp(e: Expression): Seq[Expression] = {
         val (mem, port, field, tail) = splitMemRef(e)
         // Fields that need to be replicated for each resulting mem
-        if (Seq("addr", "en", "clk", "rmode").contains(field.name)) {
+        if (Seq("addr", "en", "clk", "wmode").contains(field.name)) {
```

Comment:
this bug wouldn't have come up if these were checked by the parser, and turned into fields in IR nodes that can be statically checked by the scala compiler